### PR TITLE
[Helix] Install ef tool to work item root instead of globally

### PR DESCRIPTION
**Pull request by HaoK**

### CODE REVIEW

**Comment by HaoK** [comment]
Looks like this change is good I think, all the templates are quarantined but this change seems to be fine as the template tests passed on the quarantined run: https://dev.azure.com/dnceng/public/_build/results?buildId=608820&view=ms.vss-test-web.build-test-results-tab&runId=19116312&resultId=100126&paneView=debug

File: .azure/pipelines/ci.yml [hunk 1]
```diff
@@ -692,7 +692,7 @@ stages:
 
   - template: jobs/default-build.yml
     parameters:
-      condition: notin(variables['Build.Reason'], 'PullRequest')
+      # condition: notin(variables['Build.Reason'], 'PullRequest')
```
Comment (HaoK) [inline .azure/pipelines/ci.yml:695]: Revert before merge

**Review by dougbu — APPROVED** [review]
I guess this is fine. But, why are we installing `dotnet` anywhere outside the Helix root folder?

**Comment by HaoK** [comment]
The tool should be the only global thing we were laying down

**Comment by dougbu** [comment]
@HaoK aren't global tools installed in folders under where `dotnet` was found?

(Seems strange to install tools in the user folder even when `$env:DOTNET_ROOT` has been overridden.  But, if that's the case, carry on&hellip;)

**Comment by HaoK** [comment]
Well I'm just doing I understood the consensus on https://github.com/dotnet/aspnetcore/issues/20955 was, which is install the tool into a local folder so it has no sideeffects, doesn't seem worth the trouble doing anything else, we just need to verify the ef migration works, so installing it to the work item directory and running it from there seems always safe.

**Pull request merged**